### PR TITLE
Fix fetching volumeHandle and manually attach Volumes on AWS post detach

### LINKED ISSUE #8111: Update test_detach_attach_worker_volume and test_detach_attach_2_data_volumes for OCP 4.11 and higher versions on AWS
Change the yaml navigation for fetching VolumeID from yaml by updating it's Type
Old
```
Source:
    Type:       AWSElasticBlockStore (a Persistent Disk resource in AWS)
    VolumeID:   aws://us-east-2c/vol-0d2f121edf9dd9a41
    FSType:     
    Partition:  0
    ReadOnly:   false
```

New
```
Source:
    Type:              CSI (a Container Storage Interface (CSI) volume source)
    Driver:            ebs.csi.aws.com
    FSType:            
    VolumeHandle:      vol-04aa69be6fd5c46a9
    ReadOnly:          false
    VolumeAttributes:      storage.kubernetes.io/csiProvisionerIdentity=1687252649220-8081-ebs.csi.aws.com
```
In `get_data_volumes`  in ocs_ci/utility/aws.py

The type has changed from AWSElasticBlockStore to CSI hence the test is failing probably starting OCP 4.11
The test passes this step until OCP 4.10

### PR DESCRIPTION
**Pull request by am-agrawa**
Fixes #8111

### CODE REVIEW

**Comment by am-agrawa** [comment]
Test passed twice on 2 different runs- https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-deploy-ocs-cluster/37396/testReport/tests.functional.z_cluster.nodes.test_disk_failures/TestDiskFailures/

The initial failure seems intermittent as per dev's analysis.

**Review by prsurve — DISMISSED** [review]

File: tests/functional/z_cluster/nodes/test_disk_failures.py [hunk 1]
```diff
@@ -63,20 +66,26 @@ def detach_volume_and_wait_for_attach(self, nodes, data_volume, worker_node):
...
             """
             Wait for worker volume to be re-attached automatically
             to the node
             """
-            assert nodes.wait_for_volume_attach(data_volume), (
-                f"Volume {data_volume} failed to be re-attached to worker "
-                f"node {worker_node}"
-            )
+            logger.info(f"Volume {data_volume} is deattached successfully")
+            if config.ENV_DATA.get("platform", "").lower() == constants.AWS_PLATFORM:
```
[the remark this answers is not in the document]
Comment (am-agrawa) [reply tests/functional/z_cluster/nodes/test_disk_failures.py:79]: @ebenahar The test is failing for IBM too, so I am not 100% sure. @prsurve is working on issue #8117 which should further address this query.
Bdw, can we reach out to someone from ENG and confirm the same?

[the remark this answers is not in the document]
Comment (am-agrawa) [reply tests/functional/z_cluster/nodes/test_disk_failures.py:79]: @ebenahar Do you think we can merge it for now?

[the remark this answers is not in the document]
Comment (ebenahar) [reply tests/functional/z_cluster/nodes/test_disk_failures.py:79]: yes

**Review by prsurve — APPROVED** [review]

**Comment by am-agrawa** [comment]
4.13 backport [PR](https://github.com/red-hat-storage/ocs-ci/pull/9907)

**Comment by am-agrawa** [comment]
4.14 backport [PR](https://github.com/red-hat-storage/ocs-ci/pull/9917)

**Pull request merged**